datafusion/common/src/utils.rs

```diff
@@ -22,8 +22,16 @@ use arrow::array::ArrayRef;
 use arrow::compute::SortOptions;
 use std::cmp::Ordering;
 
+/// Given column vectors, returns row at `idx`.
+pub fn get_row_at_idx(columns: &[ArrayRef], idx: usize) -> Result<Vec<ScalarValue>> {
+    columns
+        .iter()
+        .map(|arr| ScalarValue::try_from_array(arr, idx))
+        .collect()
+}
+
 /// This function compares two tuples depending on the given sort options.
-fn compare(
+pub fn compare_rows(
```

Comment:
maybe `pub crate` would be better here unless there is some reason to export it

Reply:
This function is used both in `datafusion/common` and `datafusion/physical_expr.` Hence unfortunately `pub(crate)` wouldn't work.